### PR TITLE
Wait for OPA to load rules in tests startup

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

OPA can take some time to load rules, but our tests don't wait for OPA, and start right after the server is loaded.
Sometimes it works, but in other times the tests may fail because OPA is still loading the rules.
This PR allows to wait for OPA during the test suite startup.

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: tests/python/shared/fixtures/init.py [hunk 1]
```diff
@@ -185,12 +186,38 @@ def delete_compose_files():
...
+        sleep(1)
+
+    raise Exception(
+        "Failed to reach the server during the specified period." "Please check the configuration."
+    )
+
+
+def wait_for_opa():
+    for i in range(300):
+        logging.getLogger(__package__).debug(f"waiting for the opa to load ... ({i})")
```
Comment (nmanovic) [inline tests/python/shared/fixtures/init.py:210]: @zhiltsov-max , we have our own health check which have the logic.

File: tests/python/shared/fixtures/init.py [hunk 2]
```diff
@@ -185,12 +186,38 @@ def delete_compose_files():
...
 
 
+def wait_for_services():
+    wait_for_server()
+    wait_for_opa()
+
+
 def wait_for_server():
-    for _ in range(30):
+    for i in range(300):
```
Comment (nmanovic) [inline tests/python/shared/fixtures/init.py:195]: @zhiltsov-max , need to use our own health check.

**Comment by nmanovic** [comment]
@kirill-sizov , could you please comment?

**Comment by kirill-sizov** [comment]
> @kirill-sizov , could you please comment?

My comment is the same as yours, probably we should reuse our `GET /api/server/healthcheck` method.

I also noticed that in our docker-compose file `cvat_server` depends on `cvat_opa`, I think this is incorrect since after recent changes `cvat_opa` became depending on `cvat_server` because `cvat_opa` cannot bundle rules without `cvat_server`

**Comment by zhiltsov-max** [comment]
Changed to use the server command, as recommended in the [docs](https://opencv.github.io/cvat/docs/administration/basics/installation/#cvat-healthcheck-command).

File: tests/python/shared/fixtures/init.py [hunk 3]
```diff
@@ -205,7 +223,7 @@ def kube_restore_data_volumes():
     pod_name = _kube_get_server_pod_name()
     kube_cp(
         CVAT_DB_DIR / "cvat_data.tar.bz2",
-        f"{pod_name}:/tmp/cvat_data.tar.bz2",
+        f"{pod_name}:/tifmp/cvat_data.tar.bz2",
```
Comment (nmanovic) [inline tests/python/shared/fixtures/init.py:226]: @zhiltsov-max , is it typo?

File: tests/python/shared/fixtures/init.py [hunk 4]
```diff
@@ -185,12 +185,30 @@ def delete_compose_files():
         filename.unlink(missing_ok=True)
 
 
+def wait_for_services():
```
Comment (nmanovic) [inline tests/python/shared/fixtures/init.py:188]: Why do we need the function?

File: tests/python/shared/fixtures/init.py [hunk 5]
```diff
@@ -185,12 +185,30 @@ def delete_compose_files():
...
-        response = requests.get(get_api_url("users/self"))
-        if response.status_code == HTTPStatus.UNAUTHORIZED:
-            break
-        sleep(5)
+    for i in range(300):
+        logging.getLogger(__package__).debug(f"waiting for the server to appear ... ({i})")
+        response = requests.get(get_server_url("api/server/health/", format="json"))
+        try:
+            statuses = response.json()
+            if all(v == "working" for v in statuses.values()):
```
Comment (nmanovic) [inline tests/python/shared/fixtures/init.py:198]: @zhiltsov-max , I believe it is better just to check the response status. If it is 200, the server is fine.

File: tests/python/shared/fixtures/init.py (hunk 3 above)
Comment (zhiltsov-max) [reply tests/python/shared/fixtures/init.py]: Yes, fixed. Sometimes I occasionally can paste something with my laptop's touchpad.

File: tests/python/shared/fixtures/init.py (hunk 4 above)
Comment (zhiltsov-max) [reply tests/python/shared/fixtures/init.py:188]: We are waiting for services now, not just for the server. `wait_for_server` reflects that it is an implementation detail that everything required is available via the server. I think, the latter might be not very interesting now, as we have no other checks.

File: tests/python/shared/fixtures/init.py [hunk 6]
```diff
@@ -185,12 +185,26 @@ def delete_compose_files():
...
 
-def wait_for_server():
-    for _ in range(30):
-        response = requests.get(get_api_url("users/self"))
-        if response.status_code == HTTPStatus.UNAUTHORIZED:
-            break
-        sleep(5)
+def wait_for_services():
+    for i in range(300):
+        logging.getLogger(__package__).debug(f"waiting for the server to load ... ({i})")
```
Comment (nmanovic) [inline tests/python/shared/fixtures/init.py:190]: Probably it is better to get the logger only once.

```
logger = logging.getLogger(__package__)
```

File: tests/python/shared/fixtures/init.py (hunk 5 above)
Comment (zhiltsov-max) [reply tests/python/shared/fixtures/init.py]: Changed

File: tests/python/shared/fixtures/init.py (hunk 6 above)
Comment (zhiltsov-max) [reply tests/python/shared/fixtures/init.py]: Changed

**Review by nmanovic — APPROVED** [review]

**Pull request merged**